### ISSUE TITLE
Unable to render page when 'meta title' page config param is set

### ISSUE THREAD

**Issue by OlegIvan0v** [magento/magento2#2956]
### Preconditions

1. Magento 2

### Steps to reproduce

1. Create a custom module with following controller: 
``` php
use Magento\Framework\View\Result\PageFactory;

class SomeAction extends \Magento\Framework\App\Action\Action
{
    /**
     * @var PageFactory
     */
    protected $resultPageFactory;

    public function __construct(
        Context $context,
        PageFactory $resultPageFactory
    ) {
        parent::__construct($context);
        $this->resultPageFactory = $resultPageFactory;
    }

    public function execute()
    {
        /** $resultPage @var \Magento\Framework\View\Result\Page */
        $resultPage = $this->resultPageFactory->create();
        $resultPage->getConfig()->setMetadata('title', 'some meta title');
        return $resultPage;
    }
}
```
2. Open the created controller on storefront.

### Expected result

Page is opened, its source contains `<meta name="title" content="some meta title">`

### Actulal result

Error message: "Recoverable Error: Object of class Magento\Framework\View\Page\Title could not be converted to string..."

**Additional info**: the reason in that page config renderer tries to use page title object as meta title content (in \Magento\Framework\View\Page\Config\Renderer::processMetadataContent()).

**Comment by ghost** [comment]
Not sure if I would expect that `getMetadata();` sets the title. Maybe it is omitted in the metadata array for a special purpose?
I cannot think of any but I have not much SEO knowledge :)
I'm just using `$resultPage->getConfig()->getTitle()->set("FOO");` to set the title which works fine

**Comment by OlegIvan0v** [comment]
The problem is that we cannot set meta title of page using \Magento\Framework\View\Page\Config::setMetadata():

``` html
<meta name="title" content="Page title">
```

It's not related to page title:

``` html
<title>Page title</title>
```

We can set any other meta data, but not 'meta title' because during processing of meta data content magento invokes getter method of \Magento\Framework\View\Page\Config, which for 'title' data will be \Magento\Framework\View\Page\Config::getTitle(), which returns \Magento\Framework\View\Page\Title object, that is not converted to string.
This problem can be solved by specifying preference of \Magento\Framework\View\Page\Config\RendererInterface, but on my opinion we must be able to specify any meta content using default renderer class \Magento\Framework\View\Page\Config\Renderer.

**Comment by shiftedreality** [comment]
Correct way to setup title was specified by @xhallix 

``` php
$resultPage->getConfig()->getTitle()->set("FOO");
```

But current behavior looks like issue. Setting metadata should have the same result for any meta name. 

Created ticket MAGETWO-47807

**Comment by slackerzz** [comment]
any news about MAGETWO-47807?

**Comment by magento-engcom-team** [comment]
@volnnn, thank you for your report.
We've created internal ticket(s) MAGETWO-84539 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @volnnn. Thank you for your report. 
The issue has been fixed in magento/magento2#11368 by @slackerzz in 2.3-develop branch
Related commit(s):
 - [32e2e4de5921ecb7b13722955298e886ec2b2d59](https://github.com/magento/magento2/commit/32e2e4de5921ecb7b13722955298e886ec2b2d59)
 - [b11c7e78b24450ec4d3e73534c40ab427e786cdd](https://github.com/magento/magento2/commit/b11c7e78b24450ec4d3e73534c40ab427e786cdd)
 - [474e43cfb97617355d6d17ec7eb7639789319a7d](https://github.com/magento/magento2/commit/474e43cfb97617355d6d17ec7eb7639789319a7d)

The fix will be available with the upcoming 2.3.0 release.

**Comment by sidolov** [comment]
Hi @volnnn. Thank you for your report. 
The issue has been fixed in magento/magento2#16948 by @ronak2ram in 2.1-develop branch
Related commit(s):
 - [082aea78624be55e779067ee40c35b42d6fe4cf3](https://github.com/magento/magento2/commit/082aea78624be55e779067ee40c35b42d6fe4cf3)
 - [94452f734a27e1865dc7db378b7bc741162f4e68](https://github.com/magento/magento2/commit/94452f734a27e1865dc7db378b7bc741162f4e68)
 - [a854267c578358adf7b1bcd9d84e57c371e37b47](https://github.com/magento/magento2/commit/a854267c578358adf7b1bcd9d84e57c371e37b47)

The fix will be available with the upcoming 2.1.15 release.

**Comment by ishakhsuvarov** [comment]
@volnnn We are closing this ticket now, as issue has been fixed in 2.1, 2.2 and 2.3.

Commits for 2.1 and 2.3 are linked in the comments above, for 2.2 reference would be https://github.com/magento/magento2/pull/16333 and https://github.com/magento/magento2/pull/15532

Thanks
